provider/pkg/gen/schema.go

```diff
@@ -915,7 +916,10 @@ func (g *packageGenerator) genProperties(typeName string, typeSchema *discovery.
 		}
 
 		apiProp := resources.CloudAPIProperty{
-			Ref:                  typeSpec.Ref,
+			Ref:      typeSpec.Ref,
+			Format:   prop.Format,
+			Required: prop.Required || isRequired(prop.Description),
```

Comment:
Should we refactor `isRequired` to take the property and internally handle both `prop.Required` and `prop.Description`?